rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -2228,33 +2231,35 @@ rmw_wait(
   // signals to the upper layers that it isn't ready.  If something is ready, then we leave it as
   // a valid pointer.
 
-  bool skip_wait = check_and_attach_condition(
-    subscriptions, guard_conditions, services, clients, events, wait_set_data);
-  if (!skip_wait) {
+  {
```

Comment:
Could we add some documentation here on why we now take the lock before calling `check_and_attach_condition`?

Reply:
Done